salt/metalk8s/container-engine/containerd/installed.sls

```diff
@@ -36,12 +38,24 @@ Install containerd:
 Create containerd service drop-in:
   file.managed:
     - name: /etc/systemd/system/containerd.service.d/50-metalk8s.conf
-    - source: salt://{{ slspath }}/files/50-metalk8s.conf
+    - source: salt://{{ slspath }}/files/50-metalk8s.conf.j2
+    - template: jinja
     - user: root
     - group: root
     - mode: 0644
     - makedirs: true
     - dir_mode: 0755
+    - context:
+      environment: >-
+      {%- if proxies %}
+        NO_PROXY={{ networks.values() | join(",") }}
+        {%- if 'http' in proxies %}
+        HTTP_PROXY={{ proxies.http }}
+        {%- endif %}
+        {%- if 'https' in proxies %}
+        HTTPS_PROXY={{ proxies.https }}
+        {%- endif %}
+      {%- endif %}
     - require:
       - metalk8s_package_manager: Install containerd
 
```

Comment:
If we amended the contanerd service configs then we need a restart for our changes to take effect. no??
The way i see the sequence
- Install containerd
- patch the configuration file
Could be inverted ??

Reply:
It's not needed since in `configured` we have a `service.running` that watch this state
BUT
for me this service configuration should be in `configured` state (but not part of this PR btw)

Reply:
Yep, not really the scope of the changes, but yeah, could be improved.